test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -1946,4 +1947,12 @@ protected static SecureRandom secureRandomFips(final byte[] seed) throws NoSuchA
         secureRandomFips.setSeed(seed); // DEFAULT/BCFIPS setSeed() is non-deterministic
         return secureRandomFips;
     }
+
+    public static void safeAwait(CyclicBarrier barrier) {
```

Comment:
Does this need to be in the base class? It's only used in `FanOutListenerTests`

Reply:
Only used in `FanOutListenerTests` so far...

I plan to follow up with a bunch more usages.

Reply:
See https://github.com/elastic/elasticsearch/pull/94737